src/modules/image/image.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2016-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { handleActions, createAction, Action } from 'redux-actions';
+
+import { createSelector } from 'reselect';
+import { convertFieldWithFallback } from '../../util/convertFieldWithFallback';
+import { ReduxState } from '../../interfaces';
+import { ImageApiType, UpdatedImageMetadata } from './imageApiInterfaces';
+
+export const fetchImage = createAction('FETCH_IMAGE');
+export const setImage = createAction<ImageApiTypeRedux>('SET_IMAGE');
+export const updateImage = createAction('UPDATE_IMAGE');
+export const updateImageSuccess = createAction('UPDATE_IMAGE_SUCCESS');
+export const updateImageError = createAction('UPDATE_IMAGE_ERROR');
+export const clearUploadedImage = createAction('CLEAR_UPLOADED_IMAGE');
+
+export const actions = {
+  updateImage,
+  fetchImage,
+  setImage,
+  updateImageSuccess,
+  updateImageError,
+  clearUploadedImage,
+};
+
+export interface ReduxImageState {
+  all: {
+    [imageId: string]: ImageApiTypeRedux;
+  };
+  isSaving: boolean;
+  images?: any;
+  uploadedImage?: any;
+}
+export type ImageApiTypeRedux = ImageApiType & { language?: string };
```

Comment:
Har rekkefølge ingenting å sei i typescript? Altså at ImageApiTypeRedux brukes før den defineres?

Reply:
Hmm, det kompileres vel, så mener at det ikke har noen ting å si.
Kan uansett endre rekkefølgen for lesbarhet :smile: